### PR TITLE
Using $manage_service in zabbix::proxy

### PR DESCRIPTION
**Pull request by Samgarr**
This PR allowing use of $manage_service in zabbix::proxy class.

### CODE REVIEW

File: manifests/proxy.pp [hunk 1]
```diff
@@ -542,14 +543,24 @@
...
   # Controlling the 'zabbix-proxy' service
-  service { $proxy_service_name:
-    ensure     => running,
-    hasstatus  => true,
-    hasrestart => true,
-    require    => [
-      Package["zabbix-proxy-${db}"],
-      File[$include_dir],
-      File[$proxy_configfile_path]],
+  if $manage_service == true {
```
Comment (bastelfreak) [inline manifests/proxy.pp]: this should be enough:
```
if $manage_service {
```

**Comment by bastelfreak** [comment]
Thanks for the PR @Samgarr. Are you able to also add spec tests for this change?

**Comment by Samgarr** [comment]
I'm not sure right now, Ruby is new for me, spec tests also.

**Comment by bastelfreak** [comment]
Don't worry. Feel free to join our IRC channel #voxpupuli on freenode. We're happy to help you with the tests.

File: manifests/proxy.pp [hunk 2]
```diff
@@ -549,34 +550,54 @@
...
+        database_type     => $database_type,
+        zabbix_type       => 'proxy',
+        database_name     => $database_name,
+        database_user     => $database_user,
+        database_password => $database_password,
+        database_host     => $database_host,
+        zabbix_proxy      => $zabbix_proxy,
+        zabbix_proxy_ip   => $zabbix_proxy_ip,
+        before            => [
+          Service[$proxy_service_name],
```
Comment (bastelfreak) [inline manifests/proxy.pp]: is this the only different line? We shouldt do code duplication here. Please write Service[$proxy_service_name] into a variable and use that in the before parameter.

**Comment by Samgarr** [comment]
Rebased, spec tests added.

**Comment by bastelfreak** [comment]
Thanks!

**Pull request merged**